game/recastnavigation_bin.ksy

```diff
@@ -0,0 +1,283 @@
+meta:
+  id: recastnavigation_bin
+  file-extension: bin
+  license: CC0-1.0
+  endian: le
+doc: this spec can be used to parse recastnavigation binary files
+doc-ref: https://masagroup.github.io/recastdetour
+seq:
+  - id: header
+    type: navmesh_set_header
+  - id: tiles
+    type: navmesh_tile
+    repeat: expr
+    repeat-expr: header.num_tiles
+types:
+  navmesh_set_header:
+    seq:
+      - id: magic
+        contents: "TESM"
+      - id: version
+        type: s4
+      - id: num_tiles
+        type: s4
+      - id: params
+        type: dt_navmesh_params
+  dt_navmesh_params:
+    seq:
+      - id: orig
+        type: float3
+        doc: The world space origin of the navigation mesh's tile space. [(x, y, z)]
+      - id: tile_width
+        type: f4
+        doc: The width of each tile. (Along the x-axis.)
+      - id: tile_height
+        type: f4
+        doc: The height of each tile. (Along the z-axis.)
+      - id: max_tiles
+        type: s4
+        doc: The maximum number of tiles the navigation mesh can contain.
+      - id: max_polys
+        type: s4
+        doc: The maximum number of polygons each tile can contain.
+    doc: |
+      Configuration parameters used to define multi-tile navigation meshes.
+      The values are used to allocate space during the initialization of a navigation mesh.
+  float3:
+    seq:
+      - id: x
+        type: f4
+      - id: y
+        type: f4
+      - id: z
+        type: f4
+  dt_tile_ref:
+    seq:
+      - id: ref
+        type: u4
+  navmesh_tile_header:
+    seq:
+      - id: tile_ref
+        type: dt_tile_ref
+      - id: data_size
+        type: s4
+  dt_mesh_header:
+    seq:
+      - id: magic
+        contents: "VAND"
+        doc: Tile magic number. (Used to identify the data format.)
+      - id: version
+        type: s4
+        doc: Tile data format version number.
+      - id: x
+        type: s4
+        doc: The x-position of the tile within the dtNavMesh tile grid. (x, y, layer)
+      - id: y
+        type: s4
+        doc: he y-position of the tile within the dtNavMesh tile grid. (x, y, layer)
+      - id: layer
+        type: s4
+        doc: The layer of the tile within the dtNavMesh tile grid. (x, y, layer)
+      - id: user_id
+        type: u4
+        doc: The user defined id of the tile.
+      - id: poly_count
```

Comment:
Please consider using `num_polys`, [as per style guide](https://doc.kaitai.io/ksy_style_guide.html#attr-id).

If you care about original program's attributes, you can add `-orig-id: polyCount` to keep it documented.